project.py

```diff
@@ -46,8 +53,19 @@ def __init__(self, n_slides=0,
         # ratio of similarity between images based on which we decide if we
         # are going to save an image
         self.similarity = similarity
+        self.__func_keyword_to_function__(compare_function)
         self.load_config()
 
+    def __func_keyword_to_function__(self, keyword):
+        switcher = {
```

Comment:
@Foxiy If you'd like to you can most probably skip the `getattr` part and do directly something like

```python
switcher = {
    'dhash': self.__compute_dhash__,
    'phash': self.__compute_phash__,
    ...
}

The `switcher.get` call will then get you a default in the same way it does now so you'd do `switcher.get(keyword, self.__compute_dhash__)`.

In any case, in combination with the 'generate from string' procedure this might and might not be worth it, I am mentioning this just in case.